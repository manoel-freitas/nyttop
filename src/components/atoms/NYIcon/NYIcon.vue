<template>
  <button class="button is-white" @click="$emit('click')">
    <font-awesome-icon :icon="icon" v-bind="$attrs"></font-awesome-icon>
  </button>
</template>

<script>
export default {
  inheritAttrs: false,
  props: {
    icon: {
      type: String,
      required: true
    }
  }
}
</script>
