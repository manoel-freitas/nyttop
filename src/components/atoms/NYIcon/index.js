import NYIcon from './NYIcon'

export default NYIcon
