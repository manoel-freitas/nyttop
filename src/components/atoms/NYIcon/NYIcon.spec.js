/// <reference types="jest" />

import { shallowMount } from '@vue/test-utils'
import NYIcon from './NYIcon'
const nyIcon = shallowMount(NYIcon, {
  props: {
    icon: 'search'
  },
  stubs: ['font-awesome-icon']
})
test('NYIcon has to be mounted', () => {
  expect(nyIcon.isVueInstance()).toBeTruthy()
})

test('snapshot test', () => {
  expect(nyIcon.element).toMatchSnapshot()
})
