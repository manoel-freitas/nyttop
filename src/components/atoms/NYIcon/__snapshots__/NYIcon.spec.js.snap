// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<button
  class="button is-white"
>
  <font-awesome-icon-stub />
</button>
`;
